streamalert_cli/athena/helpers.py

```diff
@@ -57,10 +64,21 @@ def add_partition_statement(partitions, bucket, table_name):
             continue
 
         fmt_values.update(parts.groupdict())
+        partition_stmt = PARTITION_STMT.format(**fmt_values)
+
+        # If statement will be reset if it is about to reach the MAX_QUERY_LENGTH
+        if statement_len - len(partition_stmt) - len(statement) < 0:
```

Comment:
I like the idea of counting _up_ to the ceiling of max size, not down to the floor. imho the way this logic is now makes it harder to reason about.. meaning, could you do something like:

```python

init_stmt_len = len(statement[0])
statement_len = init_stmt_len

...
partition_stmt = PARTITION_STMT.format(**fmt_values)
partition_stmt_len = len(partition_stmt)

if statement_len + partition_stmt_len >= MAX_QUERY_LENGTH:
    # do the append/yield
    ...
    # reset the initial value
    statement_len = init_stmt_len

statement_len += partition_stmt_len

...
```

Reply:
Aha, I get your point and it makes sense. Will make a change.